### ISSUE TITLE
How to achieve withoutGlobalScopes() behaviour

### ISSUE THREAD

**Issue by liepumartins** [laravel-json-api/laravel#123]
For some of our models we employ global scopes through the models `boot()` method. It is convenient, because those constrains do not need to repeated all over the application. Many relationship definitions etc.

e.g.
```php
  protected static function boot(): void
    {
        parent::boot();
self::addGlobalScope('published', static function (Builder $query): void {
            $query->whereDate('from_date', '<=', Carbon::now())
                ->whereDate('to_date', '>=', Carbon::now());
        });
self::addGlobalScope('active', static function (Builder $query): void {
            $query->where('status', '=', 1);
        });
}
``` 

However JSON API faces a problem, "inactive" or "unpublished" records are not found - 404.

I do not see a way to pass `withoutGlobalScopes()` via schema. 
What would be the suggested approach? 
create a separate model without scopes for JSON API?

**Comment by lindyhopchris** [comment]
Hi! This is something I could add suppor for. In the meantime, you can implement this yourself via the `Driver` class. This isn't documented, but basically the query builder instances are created by a driver that the schema returns. This is how I override the default implementation to add support for soft deletes - I change the driver.

For example, the basic driver looks like this:
https://github.com/laravel-json-api/eloquent/blob/develop/src/Drivers/StandardDriver.php

You could write your own (it just needs to implement the interface), then overload the `driver()` method on the Eloquent schema returning your own driver:
https://github.com/laravel-json-api/eloquent/blob/52667226d6bb3b4df1390b5958cf0b86cf07a809/src/Schema.php#L359-L364

**Comment by bbprojectnet** [comment]
@lindyhopchris
I have simillar question, it is possible to modify query for some schema relationship (also for includes)? For example, i want to disable soft deletes but only for relationship in one schema (main resource should still have soft deletes enabled).

**Comment by lindyhopchris** [comment]
@bbprojectnet i might need to consider that for the future, but in the meantime an easy workaround is to add a specific relationship to the model that disables soft deletes, and then use that relationship in the JSON:API schema.

**Comment by bbprojectnet** [comment]
Yes, that's so obvious, I feel stupid for not thinking of it before, thank you very much :)
